src/test/java/com/commercetools/sync/cartdiscounts/utils/CartDiscountUpdateActionUtilsTest.java

```diff
@@ -0,0 +1,1473 @@
+    void buildChangeSortOrderUpdateAction_WithJustOldSortOrderHasNullValue_ShouldBuildUpdateAction() {
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.getSortOrder()).thenReturn(null);
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        final String newSortOrder = "0.3";
+        when(newCartDiscountDraft.getSortOrder()).thenReturn(newSortOrder);
+
+        final Optional<UpdateAction<CartDiscount>> changeChangeSortOrderUpdateAction =
+                buildChangeSortOrderUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(changeChangeSortOrderUpdateAction).contains(ChangeSortOrder.of(newSortOrder));
+    }
+
+    @Test
+    void buildChangeSortOrderUpdateAction_WithBothSortOrderHaveNullValues_ShouldNotBuildUpdateAction() {
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.getSortOrder()).thenReturn(null);
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        when(newCartDiscountDraft.getSortOrder()).thenReturn(null);
+
+        final Optional<UpdateAction<CartDiscount>> changeChangeSortOrderUpdateAction =
+                buildChangeSortOrderUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(changeChangeSortOrderUpdateAction).isNotPresent();
+    }
+
+    @Test
+    void buildChangeRequiresDiscountCodeUpdateAction_WithDifferentValues_ShouldBuildUpdateAction() {
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.isRequiringDiscountCode()).thenReturn(true);
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        when(newCartDiscountDraft.isRequiresDiscountCode()).thenReturn(false);
+
+        final Optional<UpdateAction<CartDiscount>> changeRequiresDiscountCodeUpdateAction =
+                buildChangeRequiresDiscountCodeUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(changeRequiresDiscountCodeUpdateAction).contains(ChangeRequiresDiscountCode.of(false));
+    }
+
+    @Test
+    void buildChangeRequiresDiscountCodeUpdateAction_WithSameValues_ShouldNotBuildUpdateAction() {
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.isRequiringDiscountCode()).thenReturn(true);
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        when(newCartDiscountDraft.isRequiresDiscountCode()).thenReturn(true);
+
+        final Optional<UpdateAction<CartDiscount>> changeRequiresDiscountCodeUpdateAction =
+                buildChangeRequiresDiscountCodeUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(changeRequiresDiscountCodeUpdateAction).isNotPresent();
+    }
+
+    @Test
+    void buildChangeRequiresDiscountCodeUpdateAction_WithJustNewRequiresDisCodeHasNullValue_ShouldBuildUpdateAction() {
```

Comment:
but it seems that it is also not optional from docs: https://docs.commercetools.com/http-api-projects-cartDiscounts#cartdiscountdraft

Reply:
but actually it's not optional so user should not pass null at all.